R/navs-legacy.R

```diff
@@ -105,36 +105,224 @@ navset_hidden <- function(..., id = NULL, selected = NULL,
+#' removed in a future version of \pkg{bslib}.
+#' 
+#' @examples
+#' navbar_options(position = "static-top", bg = "#2e9f7d", underline = FALSE)
+#' 
+#' @inheritParams shiny::navbarPage
+#' @param bg a CSS color to use for the navbar's background color.
+#' @param inverse Either `TRUE` for a light text color or `FALSE` for a dark
+#'   text color. If `"auto"` (the default), the best contrast to `bg` is chosen.
+#' @param underline Whether or not to add underline styling to page or navbar
+#'   links when active or focused.
+#' @param ... Additional arguments are ignored. `...` is included for future
+#'   expansion on `navbar_options()`.
+#' 
+#' @returns Returns a list of navbar options.
+#' 
+#' @export
+navbar_options <- function(
+  ...,
+  position = c("static-top", "fixed-top", "fixed-bottom"),
+  bg = NULL,
+  inverse = "auto",
+  collapsible = TRUE,
+  underline = TRUE
+) {
+  # Track user-provided arguments for print method and deprecation warnings
+  is_default <- list(
+    position = missing(position),
+    bg = missing(bg),
+    inverse = missing(inverse),
+    collapsible = missing(collapsible),
+    underline = missing(underline)
+  )
+
+  opts <- list(
+    position = rlang::arg_match(position),
+    bg = bg,
+    inverse = inverse,
+    collapsible = collapsible,
+    underline = underline
+  )
+  
+  structure(
+    dropNulls(opts),
+    class = c("bslib_navbar_options", "list"),
+    is_default = is_default,
+    waldo_opts = list(ignore_attr = TRUE)
+  )
+}
+
+navbar_options_resolve_deprecated <- function(
+  options_user = list(),
+  position = deprecated(),
+  bg = deprecated(),
+  inverse = deprecated(),
+  collapsible = deprecated(),
+  underline = deprecated(),
+  .fn_caller = "navset_bar",
+  .warn_deprecated = TRUE
+) {
+  options_old <- list(
+    position = if (lifecycle::is_present(position)) position,
+    bg = if (lifecycle::is_present(bg)) bg,
+    inverse = if (lifecycle::is_present(inverse)) inverse,
+    collapsible = if (lifecycle::is_present(collapsible)) collapsible,
+    underline = if (lifecycle::is_present(underline)) underline
+  )
+  options_old <- dropNulls(options_old)
+
+  args_deprecated <- names(options_old)
+
+  if (.warn_deprecated && length(args_deprecated)) {
+    # TODO-deprecated: (2024-12) Elevate deprecation to an error
+    lifecycle::deprecate_warn(
+      "0.9.0",
+      I(sprintf(
+        "The %s argument%s of `%s()` have been consolidated into a single `navbar_options` argument and ",
+        paste(sprintf("`%s`", args_deprecated), collapse = ", "),
+        if (length(args_deprecated) > 1) "s" else "",
+        .fn_caller
+      ))
+    )
+  }
+  
+  # Consolidate `navbar_options` (options_user) with direction options taking
+  # the direct option if the user option is a default value, warning if
+  # otherwise ignored.
+  # TODO-deprecated: Remove this and warning when direct options are hard-deprecated
```

Comment:
```suggestion
  # Consolidate `navbar_options` (options_user) with the deprecated direct options.
  # We take the direct option if the user option is a default value, warning if
  # otherwise ignored.
  # TODO-deprecated: Remove this and warning when direct options are hard-deprecated
```